File: smelly/duplicate-assert/CSharpXUnit/CSharpXUnit.Tests/Tests/AsyncHelperTests_ContinuationTimeoutNotHitTest.cs
using Xunit;

namespace nlog
{
    public class AsyncHelperTests
    {
        [Fact]
        public void ContinuationTimeoutNotHitTest()
        {
            {
                var exceptions = new List<Exception>();
                // set up a timer to strike 
                var cont = AsyncHelpers.WithTimeout(AsyncHelpers.PreventMultipleCalls(exceptions.Add), TimeSpan.FromMilliseconds(50));
                // call success quickly, hopefully before the timer comes
                cont(null);
                // sleep to make sure timer event comes
                Thread.Sleep(100);
                // make sure we got success, not a timer exception
                Assert.Single(exceptions);
                Assert.Null(exceptions[0]);
                // those will be ignored
                cont(null);
                cont(new ApplicationException("Some exception"));
                cont(null);
                cont(new ApplicationException("Some exception"));
                Assert.Single(exceptions);
                Assert.Null(exceptions[0]);
            }
        }
    }
}

File: refactored/duplicate-assert/python-pytest/AsyncHelperTests_ContinuationTimeoutNotHitTest.py
import pytest

class TestAsyncHelper:

    def test_continuation_timeout_not_hit_test_1(self):
        exceptions = []
        cont = AsyncHelpers.with_timeout(AsyncHelpers.prevent_multiple_calls(exceptions.append), 50)
        cont(None)
        time.sleep(0.1)
        assert len(exceptions) == 1
        assert exceptions[0] is None

    def test_continuation_timeout_not_hit_test_2(self):
        exceptions = []
        cont = AsyncHelpers.with_timeout(AsyncHelpers.prevent_multiple_calls(exceptions.append), 50)
        cont(None)
        time.sleep(0.1)
        cont(None)
        cont(ApplicationException("Some exception"))
        cont(None)
        cont(ApplicationException("Some exception"))
        assert len(exceptions) == 1
        assert exceptions[0] is None
